### PR TITLE
Add HiveCacheContext into the CacheContext

### PR DESCRIPTION
**Pull request by BlueStalker**
### What changes are proposed in this pull request?

This PR add HiveCacheContext into CacheContext

### Why are the changes needed?
HiveCacheContext contains the hive-related information like schema, table, partition, this information later will be used to identify whether the corresponding file needs to be cached and its caching TTL.

### Does this PR introduce any user facing changes?
N/A

### CODE REVIEW

File: core/common/src/main/java/alluxio/client/file/CacheContext.java [hunk 1]
```diff
@@ -28,6 +29,10 @@
   private CacheQuota mCacheQuota = CacheQuota.UNLIMITED;
   /** Used in Prestodb to indicate the cache scope. */
   private CacheScope mCacheScope = CacheScope.GLOBAL;
+
+  /** Used in Prestodb to indicate the hiveContext for a file. */
+  private HiveCacheContext mHiveCacheContext = null;
```
Comment (beinan) [inline core/common/src/main/java/alluxio/client/file/CacheContext.java:34]: Might be out of scope of this PR, I'm just thinking mCacheQuota, mCacheScope and mHiveCacheContext should be final and also should be added as the parameter of the constructor.  @apc999 what do you think?

Comment (beinan) [inline core/common/src/main/java/alluxio/client/file/CacheContext.java:34]: Add @Nullable?

File: core/common/src/main/java/alluxio/client/hive/HiveCacheContext.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+/**
+ * Data structure that stores and returns Hive related Cache Scope
+ */
+public class HiveCacheContext
+{
+  private final String mDatabase;
+  private final String mTable;
+  private final String mPartition;
+
+  public HiveCacheContext(String database, String table, String partition)
```
Comment (beinan) [inline core/common/src/main/java/alluxio/client/hive/HiveCacheContext.java:26]: The partition might be null I think in the hive context.  So put an annotation @Nullable?

**Review by beinan — COMMENTED** [review]
Thank you for the contribution! just a couple of minor issues

File: core/common/src/main/java/alluxio/client/hive/HiveCacheContext.java (hunk 2 above)
Comment (BlueStalker) [reply core/common/src/main/java/alluxio/client/hive/HiveCacheContext.java:26]: sg

File: core/common/src/main/java/alluxio/client/file/CacheContext.java (hunk 1 above)
Comment (BlueStalker) [reply core/common/src/main/java/alluxio/client/file/CacheContext.java:34]: This makes sense to me and we can make default constructor

Comment (BlueStalker) [reply core/common/src/main/java/alluxio/client/file/CacheContext.java]: it is in the getter?

**Review by apc999 — APPROVED** [review]

**Comment by apc999** [comment]
the failing test "alluxio.client.rest.S3ClientRestApiTest.listAllMyBuckets" is unrelated and has been fixed in master branch

**Comment by apc999** [comment]
alluxio-bot, merge this please

**Pull request merged**